test/transaction_test.exs

```diff
@@ -98,6 +99,44 @@ defmodule TransactionTest do
     assert query("SELECT 42", []) == [[42]]
   end
 
+  @tag mode: :transaction
+  @tag disconnect_on_read_only_transaction_error: true
+  test "transaction read-only only error disconnects with prepare and execute", context do
+    Process.flag(:trap_exit, true)
+
+    assert transaction(fn conn  ->
+      P.query!(conn, "SET TRANSACTION READ ONLY", []).connection_id
+
+      {:ok, query} = P.prepare(conn, "query_1", "insert into uniques values (1);", [])
+
+      assert capture_log(fn ->
+        {:error, %Postgrex.Error{postgres: %{message: "cannot execute INSERT in a read-only transaction"}}} =
```

Comment:
@wojtekmach we should match on the code rather than on the message because the message can be internationalized. This will also fix the build on 8.4.